viz_scripts/plots.py

```diff
@@ -48,82 +57,148 @@ def merge_small_entries(labels, values):
-    all_labels= ['IN_VEHICLE',
-                 'UNKNOWN',
-                 'WALKING',
-                 'AIR_OR_HSR',
-                 'BICYCLING',
-                 'OTHER']
-
-    val2labeldf = pd.DataFrame({"labels": labels, "values": values})
-
-    colours = dict(zip(all_labels, plt.cm.tab10.colors[:len(all_labels)]))
-    fig, ax = plt.subplots(figsize=(10, 10), subplot_kw=dict(aspect="equal"))
-
-    m_labels, m_values = merge_small_entries(labels, values)
+    return (v2l_df.index.to_list(),v2l_df.vals.to_list(), v2l_df.pct.to_list())
 
-    wedges, texts, autotexts = ax.pie(m_values,
-                                      labels = m_labels,
-                                      colors=[colours[key] for key in labels],
-                                      pctdistance=0.75,
-                                      autopct= lambda pct: format_pct(pct, values),
-                                      textprops={'size': 23})
-
-    ax.set_title(plot_title, size=25)
-    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
-    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
-    plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
-    plt.show()
-
-def pie_chart_purpose(plot_title,labels,values,colors_map,file_name):
-    
-    fig, ax = plt.subplots(figsize=(10, 10), subplot_kw=dict(aspect="equal"))
-
-    m_labels, m_values = merge_small_entries(labels, values)
-    
-    def func(pct, values):
-        total = sum(values)
-        absolute = int(round(pct*total/100.0))
-        return "{:.1f}%\n({:d})".format(pct, absolute) if pct > 3 else''
-    
-    wedges, texts, autotexts = ax.pie(m_values,
-                                      labels = m_labels,
-                                      colors=[colors_map[key] for key in labels],
-                                      pctdistance=0.85,
-                                      autopct=lambda pct: func(pct, values),
-                                      textprops={'size': 23})
-
-    ax.set_title(plot_title, size=25)
-    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
-    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
+def process_distance_data(df, df_col, distance_col, label_units_lower):
+    """ Inputs:
+    df = Likely expanded_ct, data_eb or expanded_ct_sensed data frame
+    df_col = Column from the above df, likely Mode_confirm, primary_mode
+    distance_col = Column associated with distance from above data frame
+    label_units_lower = lbs/kg
+    trip_type = Bar labels (e.g. Labeled by user (Confirmed trips))
+    """
+    try:
+        dist = df.groupby(df_col).agg({distance_col: ['sum', 'count', 'mean']})
+        dist.columns = ['Total (' + label_units_lower + ')', 'Count', 'Average (' + label_units_lower + ')']
+        dist = dist.reset_index()
+        dist = dist.sort_values(by=['Total (' + label_units_lower + ')'], ascending=False)
```

Comment:
I spent some time looking at this, and it is basically aggregating by sum instead of count.
I don't even see why we are aggregating by count and mean here given that we don't use the mean.
We only use the mean in the average trip length plot, which is not this one.